libsrc/lynx/ser/lynx-comlynx.s

```diff
@@ -218,31 +184,43 @@ baudsuccess:
         beq     checkhs
         cmp     #SER_PAR_SPACE
         bne     @L0
-        ldx     #TxOpenColl
+        ldx     #TXOPEN
         stx     contrl
         bra     checkhs
 @L0:
-        ldx     #TxParEnable|TxOpenColl|ParEven
+        ldx     #PAREN|TXOPEN|PAREVEN
         stx     contrl
         cmp     #SER_PAR_EVEN
         beq     checkhs
-        ldx     #TxParEnable|TxOpenColl
+        ldx     #PAREN|TXOPEN
         stx     contrl
 checkhs:
         ldx     contrl
         stx     SERCTL
         ldy     #SER_PARAMS::HANDSHAKE  ; Handshake
         lda     (ptr1),y
         cmp     #SER_HS_NONE
+        beq     redeye_ok
+        cmp     #SER_HS_SW              ; Software handshake will check for connected redeye
         bne     invparameter
+
+        lda     IODAT
+        and     #NOEXP                  ; Check if redeye bit flag is unset
+        beq     redeye_ok
+        lda     #SER_ERR_NO_DEVICE      ; ComLynx cable is not inserted
+        ldx     #0
+        rts
+
+redeye_ok:
         lda     SERDAT
         lda     contrl
-        ora     #RxIntEnable|ResetErr
+        ora     #RXINTEN|RESETERR       ; Turn on interrupts for receive
         sta     SERCTL
         lda     #SER_ERR_OK
         .assert SER_ERR_OK = 0, error
         tax
```

Comment:
move tax up directly after redeye_ok, no assert needed

Reply:
The return code for the ser_open is a char, so the accumulator needs to contain the error code (SER_ERR_OK when there is no error), and x should be 0. At this point, a ldx #$00 would be okay, but since SER_ERR_OK is asserted to be zero (first number in enum values of SER_ERR_*) it saves a byte by doing tax. 
This code is similar to other serial driver implementations.

Reply:
yes, but A is zero before "lda SERDAT", so it makes sense to move it there. and get rid of the assert.

Reply:
Okay to resolve?

Reply:
well, I see no sense in polluting code with asserts which are not needed. No dead code if you can avoid it.
But just my 2cts.

Reply:
Agree. 
In this case I would follow @mrdudz recommendation of the contribution guidelines and leave it.

Reply:
but why not
```
.if SER_ERR_OK != 0
  ldx #0
 .endif
```